nss-defective-foxes/nutshell
# Initial commit, API manager functional

## Link to Ticket
Closes # [Link](https://github.com/nss-defective-foxes/nutshell/issues/8)

## Description of Proposed Changes

- Added API manager to perform CRUD operations

## Steps to Test

Outline the steps to test

```sh
git fetch --all
git checkout KFAPIManager
```
-Run JSON server using port 8088
-Run the following code in main to ensure that the API Manager is correctly performing CRUD operations

```
const APIManager = require("./api/APIManager")

APIManager.getAllUsers().then(result => console.log(result))

APIManager.getSubsetUsers(1).then(result => console.log(result))

APIManager.putMessage(1, {
id: 1,
userID: 2,
message: "message 1 by user 2"
}).then(result => console.log(result))

APIManager.deleteUser(1).then(result => console.log(result))

APIManager.postUser({
            username: "AnEvilHerbivore",
            email: "katheryn@nutshell.com",
            password: ""
        }).then(result => console.log(result))
```

Check JSON file after running each function to ensure the correct changes are being made.

## Impacted Areas in Application

List general components of the application that this PR will affect:

*Data

## Mentions @username

Tag users that need to review this code

## Definition of Done

1. The project must be fully documented. This includes the following:
   1. Complete README that documents the steps to install the code, how to install any dependencies, or system configuration needed.
   1. Every class must be documented with purpose, author, and methods.
   1. Every method must be documented with purpose and argument list - which itself must contain a short purpose for each argument.
1. The project must be able to run fully, and without errors, on each teammate's system.
1. Fulfills every requirement.
Every line of code has been peer reviewed.
1. For projects that require unit testing, core functionality must be identified and have at least one test for each
